contents/julia_101.md

```diff
@@ -423,55 +517,2108 @@ logarithm(10)
+
+#### Date Operations {#sec:dates_operations}
+
+We can perform **operations** in `Dates` instances.
+For example, we can add days to a `Date` or `DateTime` instance.
+Notice that Julia's `Dates` will automatically perform the adjustments necessary for leapyears, of months with 30 or 31 days (this is known as *calendrical* arithmetic).
+
+```jl
+sco(
+"""
+my_birthday + Day(90)
+"""
+)
+```
+
+We can add as many as we like:
+
+```jl
+sco(
+"""
+my_birthday + Day(90) + Month(2) + Year(1)
+"""
+)
+```
+
+To get **date duration**, we just use the **subtraction** `-` operator.
+Let's see how many days Jose is old:
+
+```jl
+sco(
+"""
+today() - my_birthday
+"""
+)
+```
+
+The **default duration** of `Date` types is a `Day` instance.
+For the `DateTime`, the default duration is `Millisecond` instance:
+
+```jl
+sco(
+"""
+DateTime(today()) - DateTime(my_birthday)
+"""
+)
+```
+
+#### Date Intervals {#sec:dates_intervals}
+
+One nice thing about `Dates` module is that we can also easily construct date and time intervals.
+For example suppose you want to create a `Day` interval.
+This is easy done with the colon `:` operator:
+
+```jl
+sco(
+"""
+Date("2021-01-01"):Day(1):Date("2021-01-07")
+"""
+)
+```
+
+There is nothing special in using `Day(1)` as interval, we can **use whatever `Period` type** as interval.
+For example using 3 days as intervals:
+
+```jl
+sco(
+"""
+Date("2021-01-01"):Day(3):Date("2021-01-07")
+"""
+)
+```
+
+Or even months:
+
+```jl
+sco(
+"""
+Date("2021-01-01"):Month(1):Date("2021-03-01")
+"""
+)
+```
+
+Note that the **type of this interval is a `StepRange` with the `Date` and concrete `Period` type** we used as interval inside the colon `:` operator:
+
+```jl
+sco(
+"""
+my_date_interval = Date("2021-01-01"):Month(1):Date("2021-03-01")
+typeof(my_date_interval)
+"""
+)
+```
+
+We can convert this to a **vector** with the `collect` function:
+
+```jl
+sco(
+"""
+my_date_interval_vector = collect(my_date_interval)
+"""
+)
+```
+
+And have all the **array functionalities available**, like, for example, indexing:
+
+```jl
+sco(
+"""
+my_date_interval_vector[end]
+"""
+)
+```
+
+We can also **broadcast date operations** to our vector of `Date`s:
+
+```jl
+sco(
+"""
+my_date_interval_vector .+ Day(10)
+"""
+)
+```
+
+All we've done with `Date` types can be extended to `DateTime` types in the same manner.
+
+### Random Numbers {#sec:random}
+
+Another important module in Julia's standard library is the `Random` module.
+This module deals with **random number generation**.
+`Random` is a rich library and, if you interested in it, you should consult [Julia's `Random` documentation](https://docs.julialang.org/en/v1/stdlib/Random/).
+We will cover *only* three functions: `seed!`, `rand` and `randn`.
+
+To begin we first import the `Random` module:
+
+```julia
+using Random
+```
+
+We have **two main functions that generate random numbers**:
+
+* `rand`: samples a **random element** of a data structure or type.
```

Comment:
Didn't you say that `rand` is in Julia Base? I can use it without using Random, and it doesn't even load more methods
```
julia> rand
rand (generic function with 69 methods)

julia> using Random

julia> rand
rand (generic function with 69 methods)
```

Reply:
Done in d728d40